collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -762,7 +762,7 @@ protected void onActivityResult(int requestCode, int resultCode, final Intent in
 
         // If we're coming back from the hierarchy view, the user has either tapped the back
         // button or another question to jump to so we need to rebuild the view.
-        if (requestCode == RequestCodes.HIERARCHY_ACTIVITY) {
+        if (requestCode == RequestCodes.HIERARCHY_ACTIVITY || requestCode == RequestCodes.CHANGE_SETTINGS) {
```

Comment:
This isn't tested. I played around with various Espresso tests but couldn't get something that worked nicely and actually drove this out. We still consider `FormEntryActivity` a no-go for Robolectric testing, so I didn't even look at that. 

The new [Activity Result APIs](https://developer.android.com/training/basics/intents/result) are really the answer here I think, but they are still in beta. Once that's stable we should be able to move our activity result code into components (like `FormEntryMenuDelegate`) and test the request and "response" in one place.